sklearn/svm/classes.py

```diff
@@ -158,6 +159,15 @@ def __init__(self, penalty='l2', loss='squared_hinge', dual=True, tol=1e-4, C=1.
         self.verbose = verbose
         self.random_state = random_state
         self.max_iter = max_iter
+        # FIXME Remove l1/l2 support in 1.0?
+        # FIXME Remove case insensitivity in 0.18
+        if self.loss.lower() in ('l1', 'l2'):
+            loss = {'l1': 'hinge',
+                    'l2': 'squared_hinge'}.get(self.loss.lower())
+            warnings.warn(("loss='l1' (resp. loss='l2') is deprecated and will"
+                           "be removed before version 1.0. Please use loss="
+                           "'hinge' (resp. loss='squared_hinge') instead"),
+                          DeprecationWarning)
```

Comment:
> Please also add backward compat for penalty in ('L1', 'L2') (uppercase) 

this is done in [this](https://github.com/scikit-learn/scikit-learn/pull/4261/files#diff-0093954a82eb61ab603cf6de5b0758eaR609) line...

> And add deprecation warning tests for those.

okay!